chainio/clients/avsregistry/reader_test.go

```diff
@@ -0,0 +1,158 @@
+package avsregistry
+
+import (
+	"context"
+	"log/slog"
+	"math/big"
+	"os"
+	"testing"
+
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	"github.com/Layr-Labs/eigensdk-go/testutils"
+	"github.com/Layr-Labs/eigensdk-go/types"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/stretchr/testify/require"
+)
+
+func TestReaderMethods(t *testing.T) {
+	anvilStateFileName := "contracts-deployed-anvil-state.json"
+	anvilC, err := testutils.StartAnvilContainer(anvilStateFileName)
+	require.NoError(t, err)
+
+	anvilHttpEndpoint, err := anvilC.Endpoint(context.Background(), "http")
+	require.NoError(t, err)
+
+	contractAddrs := testutils.GetContractAddressesFromContractRegistry(anvilHttpEndpoint)
```

Comment:
Think it might be worth refactoring out the test setup code into a helper that can be used across chainio/clients - seeing similar setups needed in https://github.com/Layr-Labs/eigensdk-go/pull/361/files

Reply:
Made a refactor and extracted the anvil container and clients build to use in both AvsRegistry Writer and Reader tests! We can use it also in the [ElContracts tests PR](https://github.com/Layr-Labs/eigensdk-go/pull/361/files)